connectors/spark/src/main/scala/io/unitycatalog/spark/utils/Clock.java

```diff
@@ -0,0 +1,55 @@
+package io.unitycatalog.spark.utils;
+
+import java.time.Duration;
+import java.time.Instant;
+
+public interface Clock {
+  /**
+   * @return the current time of the clock.
+   */
+  Instant now();
+
+  /**
+   * Advances the current time of this clock by the specified duration. After this call,
+   * {@link #now()} should return a time equal to the previous time plus the given {@code duration}.
+   */
+  void advance(Duration duration);
+
+  static Clock systemClock() {
+    return new SystemClock();
+  }
+
+  static Clock manualClock(Instant now) {
+    return new ManualClock(now);
+  }
+
+  class SystemClock implements Clock {
+    @Override
+    public Instant now() {
+      return Instant.now();
+    }
+
+    @Override
+    public void advance(Duration duration) {
+      throw new UnsupportedOperationException("Cannot advance system clock.");
+    }
+  }
+
+  class ManualClock implements Clock {
+    private volatile Instant now;
+
+    ManualClock(Instant now) {
+      this.now = now;
+    }
+
+    @Override
+    public synchronized Instant now() {
+      return now;
+    }
+
+    @Override
+    public synchronized void advance(Duration duration) {
```

Comment:
Need to add the `synchronized` here, since it's possible for us to advance the same `now` at the same time via different threads.  Since `Instant` is immutable, we always need to assign the updated instant to `now`.